python/sbp/client/drivers/base_driver.py

```diff
@@ -8,59 +8,61 @@
 # EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
 # WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
 
+
 class BaseDriver(object):
-  """
-  BaseDriver
+    """
+    BaseDriver
 
-  The :class:`BaseDriver` class wraps IO sources of SBP messages and provides
-  context management.
+    The :class:`BaseDriver` class wraps IO sources of SBP messages and provides
+    context management.
 
-  Parameters
-  ----------
-  handle : port
-    Stream of bytes to read from and write to.
-  """
-  def __init__(self, handle):
-    self.handle = handle
+    Parameters
+    ----------
+    handle : port
+      Stream of bytes to read from and write to.
+    """
 
-  def __enter__(self):
-    self.flush()
-    return self
+    def __init__(self, handle):
+        self.handle = handle
 
-  def __exit__(self, *args):
-    self.flush()
-    self.close()
+    def __enter__(self):
+        self.flush()
+        return self
 
-  def read(self, size):
-    """
-    Read wrapper.
+    def __exit__(self, *args):
+        self.flush()
+        self.close()
 
-    Parameters
-    ----------
-    size : int
-      Number of bytes to read.
-    """
-    return self.handle.read(size)
+    def read(self, size):
+        """
+        Read wrapper.
 
-  def write(self, s):
-    """
-    Write wrapper.
+        Parameters
+        ----------
+        size : int
+          Number of bytes to read.
+        """
+        return self.handle.read(size)
```

Comment:
One thing to consider (most likely in a separate PR though) is to do the reads in bigger chunks into a internal memory buffer. This helps with CPU load at least with serial ports. Reading in too big blocks will cause noticeable delays though.